packages/grafana-data/src/datetime/durationutil.ts

```diff
@@ -46,7 +46,8 @@ export function parseDuration(duration: string): Duration {
       return acc;
     }
 
-    const key = Object.entries(durationMap).find(([_, abbreviations]) => abbreviations?.includes(match[2]))?.[0];
+    const durationMapKey = Object.entries(durationMap).find(([_, abbreviations]) => abbreviations?.includes(match[2]));
+    const key = Number(match[1]) !== 1 ? durationMapKey?.[1][3] : durationMapKey?.[1][2];
```

Comment:
I'm afraid this might cause some breaking changes in the codebase. I think it would make more sense if we format the output instead in the refreshPicker

Reply:
@thisisobate I wouldn't bother changing it tbh. 1 days is good enough for now. We'll need to think about pluralisation properly when we do the i18n work so this would have to change anyway.

Reply:
oh...just saw your comment now. Already pushed some changes that formats the units. Please take a look and see if we need to go ahead with that approach or not